app/foi/models.py

```diff
@@ -0,0 +1,126 @@
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.text import slugify
+from wagtail.admin.panels import FieldPanel, MultiFieldPanel
+from wagtail.api import APIField
+from wagtail.coreutils import find_available_slug
+from wagtail.fields import StreamField
+from wagtail.search import index
+
+from app.core.models import BasePage, BasePageWithRequiredIntro
+
+from .blocks import AnnexeStreamBlock, RequestStreamBlock, ResponseStreamBlock
+
+
+class FoiIndexPage(BasePageWithRequiredIntro):
+    class Meta:
+        verbose_name = "Freedom of information listing page"
+
+    max_count = 1
+    subpage_types = ["foi.FoiRequestPage"]
```

Comment:
Is there a specific page type this is coming under? If we leave `parent_page_types` empty then it can be created under any page type that has no `subpage_types` set. So right now I can create these under a `HubPage`, `GeneralPage`, `HomePage` - etc. Obviously the `max_count` limits this but still possible (if the page got deleted, for example)

Reply:
I don't know yet but I imagine it's going to either be under a `HubPage` or `GeneralPage`. I thought it would be easiest to leave it open for now because I don't _think_ changing this changes the model?

Reply:
Cool - yep it's just Wagtail validation (not DB) so that's fine 👌